mautic/mautic
# Fix campaign event for condition "Contact field value" with utm source

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | N
| Automated tests included? | Y
| Related user documentation PR URL | N/A
| Related developer documentation PR URL | N/A 
| Issues addressed (#s or URLs) | N/A
| BC breaks? | N/A
| Deprecations? | N/A

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
While executing a campaign with condition "Contact field value" and choosen contact field as "UTM Source" we are getting following error when campaign is executing:
```
CustomFieldHelper.php line 60:
  Type error: Argument 1 passed to Mautic\LeadBundle\Helper\CustomFieldHelper  
  ::fieldValueTransfomer() must be of the type array, null given
```

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Create a campaign with segment as a source.
2. Add condition "Contact field value" and choose contact field as "UTM Source".
3. Add contact which will get added to Campaign's source segment.
4. On campaign execution you will get error In 
```
CustomFieldHelper.php line 60:
  Type error: Argument 1 passed to Mautic\LeadBundle\Helper\CustomFieldHelper  
  ::fieldValueTransfomer() must be of the type array, null given
```

#### Steps to test this PR:
1. Create a campaign with segment as a source.
2. Add condition "Contact field value" and choose contact field as "UTM Source".
3. Add contact which will get added to Campaign's source segment.
4. Campaign should execute successfully.

#### Other areas of Mautic that may be affected by the change:
1. Campaign event execution

#### List deprecations along with the new alternative:
1. N/A

#### List backwards compatibility breaks:
1. N/A
